catanatron_core/catanatron/state.py

```diff
@@ -214,6 +218,10 @@ def copy(self):
         state_copy.is_road_building = self.is_road_building
         state_copy.free_roads_available = self.free_roads_available
 
+        state_copy.is_resolving_trade = self.is_resolving_trade
+        state_copy.current_trade = self.current_trade
+        state_copy.acceptees = self.acceptees
+
```

Comment:
should there be a source of truth about what these attributes to care about are? The reason I ask is because when doing equality checks, etc, it will be useful to have which attributes matter in a state. I also wonder if this state should be a dataclass with an `__eq__()` predefined.

Reply:
What do you mean by a source of truth? How would that look like?

Re: DataClass, as long as it doesn't slow down state-copying we can consider it. So far it seems the closer we get this class to a `dict` or an `list` of primitives, the better for performance.